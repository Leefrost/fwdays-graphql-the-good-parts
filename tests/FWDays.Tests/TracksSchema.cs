using HotChocolate;
using Microsoft.Extensions.DependencyInjection;
using Microsoft.EntityFrameworkCore;
using HotChocolate.Execution;
using Snapshooter.Xunit;
using FWDays.Tracks.Database;

using Mutations = FWDays.Tracks.Mutations;
using Queries = FWDays.Tracks.Queries;

namespace FWDays.Tests;

public class TracksSchema
{
    [Fact]
    public async Task SchemaChanged()
    {
        ISchema schema =
            await new ServiceCollection()
                .AddDbContextPool<TracksDbContext>(
                    options => options.UseInMemoryDatabase("Data Source=fwdays.db"))
                .AddGraphQL()
            .AddQueryType(d => d.Name("Query"))
                .AddTypeExtension<Queries>()
            .AddMutationType(d => d.Name("Mutation"))
                .AddTypeExtension<Mutations>()
            .BuildSchemaAsync();

        schema.Print().MatchSnapshot();
    }
}