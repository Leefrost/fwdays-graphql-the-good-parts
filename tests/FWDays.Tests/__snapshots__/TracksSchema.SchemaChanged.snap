schema {
  query: Query
  mutation: Mutation
}

type AssignParticipantPayload {
  trackId: Int!
  participant: String!
}

type CreateTrackPayload {
  trackId: Int!
  topic: String!
  speaker: String!
}

type Mutation {
  createTrack(input: TrackInput!): CreateTrackPayload!
  assignParticipant(trackId: Int! participantInput: TrackParticipantInput!): AssignParticipantPayload!
}

"Information about pagination in a connection."
type PageInfo {
  "Indicates whether more edges exist following the set defined by the clients arguments."
  hasNextPage: Boolean!
  "Indicates whether more edges exist prior the set defined by the clients arguments."
  hasPreviousPage: Boolean!
  "When paginating backwards, the cursor to continue."
  startCursor: String
  "When paginating forwards, the cursor to continue."
  endCursor: String
}

type Query {
  tracks("Returns the first _n_ elements from the list." first: Int "Returns the elements in the list that come after the specified cursor." after: String "Returns the last _n_ elements from the list." last: Int "Returns the elements in the list that come before the specified cursor." before: String): TracksConnection
}

type Track {
  id: Int!
  topic: String
  speaker: TrackSpeaker!
  participants: [TrackParticipant!]!
}

type TrackParticipant {
  id: Int!
  firstName: String!
  lastName: String!
}

type TrackSpeaker {
  id: Int!
  firstName: String!
  lastName: String!
}

"A connection to a list of items."
type TracksConnection {
  "Information to aid in pagination."
  pageInfo: PageInfo!
  "A list of edges."
  edges: [TracksEdge!]
  "A flattened list of the nodes."
  nodes: [Track!]
}

"An edge in a connection."
type TracksEdge {
  "A cursor for use in pagination."
  cursor: String!
  "The item at the end of the edge."
  node: Track!
}

input TrackInput {
  topic: String!
  speaker: TrackSpeakerInput!
}

input TrackParticipantInput {
  firstName: String!
  lastName: String!
}

input TrackSpeakerInput {
  firstName: String!
  lastName: String!
}
